cli/src/utilities/publishToGitHub.ts

```diff
@@ -0,0 +1,162 @@
+import { GluegunToolbox } from 'gluegun';
+import { execSync } from 'child_process';
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const verifyRepoDeleted = (repoName: string): boolean => {
+  try {
+    execSync(`gh repo view ${repoName}`, { stdio: 'ignore' });
+    return false; // Repository still exists
+  } catch (e) {
+    return true; // Repository is deleted
+  }
+};
+
+export const publishToGitHub = async (toolbox: GluegunToolbox, projectName: string): Promise<string | null> => {
+  const {
+    print: { info, error }
+  } = toolbox;
+
+  try {
+    // Check if gh CLI is installed
+    try {
+      execSync('gh --version', { stdio: 'ignore' });
+    } catch (e) {
+      error('\nGitHub CLI (gh) is not installed.');
+      info('Please install it from: https://cli.github.com/');
+      info('Then run: gh auth login');
+      return null;
+    }
+
+    // Check if user is authenticated with correct permissions
+    try {
+      const scopes = execSync('gh auth status -t 2>&1', { encoding: 'utf8' });
```

Comment:
Also -- should we only check for the `delete_repo` permission when we go to delete a repository?

Reply:
i don't quite follow, will merge for now though